### PR TITLE
Support Hot Module Replacement using Vite

### PR DESCRIPTION
**Pull request by andychow326**
ref DEV-2008

### CODE REVIEW

File: authui/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "prettier": "prettier --cache --list-different ./src ./packages",
     "eslint": "eslint --cache './src/**/*.{js,ts}'",
     "stylelint": "stylelint --cache './src/**/*.{css,scss}'",
+    "dev": "vite",
```
Comment (louischan-oursky) [inline authui/package.json:17]: `make authui-dev`

File: authui/packages/vite-plugin-authgear-authui/src/index.js [hunk 2]
```diff
@@ -99,13 +102,98 @@ async function writeHTMLTemplate(targetPath, templateName, elements) {
...
+    const root = htmlParser.parse(source);
+    const head = root.getElementsByTagName("head")[0];
+
+    const nodes = [];
+    for (const node of head.childNodes) {
+      if (node.attributes == null) {
+        continue;
+      }
+
+      // Rewrite node attributes by condition
```
Comment (louischan-oursky) [inline authui/packages/vite-plugin-authgear-authui/src/index.js:125]: if this is `script src`
if this is `link rel=stylesheet href`

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
SCR-20240906-kvcq
SCR-20240906-kviz

Looks like you did not handle other kind of asset properly.

**Comment by andychow326** [comment]
> SCR-20240906-kvcq SCR-20240906-kviz
> 
> Looks like you did not handle other kind of asset properly.

@louischan-oursky It looks like there are 2 problems to be fixed:
- Seems it is required to run `make authui` before `make authui-dev` as there are some other places using `call $.GeneratedStaticAssetURL` other than `__generated_asset.html`. (Fixed by [c28b3dc](https://github.com/authgear/authgear-server/pull/4706/commits/c28b3dc8767ad628971879c9b225d263050b2821))
- Noticed that `Twemoji.Mozilla-subset.woff2` still cannot be imported even the above is fixed. It is because the hash removing process for `Twemoji.Mozilla-subset.woff2` is not correctly handled. (Fixed by [94d8b45](https://github.com/authgear/authgear-server/pull/4706/commits/94d8b45f9ab11c6d87f82498997364729a80d5f7))

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**